src/runtime/objmodel.cpp

```diff
@@ -2276,19 +2276,14 @@ static std::string getFunctionName(CLFunction* f) {
     return "<unknown function>";
 }
 
-static void placeKeyword(const std::vector<AST_expr*>& arg_names, std::vector<bool>& params_filled,
-                         const std::string& kw_name, Box* kw_val, Box*& oarg1, Box*& oarg2, Box*& oarg3, Box** oargs,
-                         BoxedDict* okwargs, CLFunction* cl) {
+static void placeKeyword(const ParamNames& param_names, std::vector<bool>& params_filled, const std::string& kw_name,
+                         Box* kw_val, Box*& oarg1, Box*& oarg2, Box*& oarg3, Box** oargs, BoxedDict* okwargs,
+                         CLFunction* cl) {
     assert(kw_val);
 
     bool found = false;
-    for (int j = 0; j < arg_names.size(); j++) {
-        AST_expr* e = arg_names[j];
-        if (e->type != AST_TYPE::Name)
-            continue;
-
-        AST_Name* n = ast_cast<AST_Name>(e);
-        if (n->id == kw_name) {
+    for (int j = 0; j < param_names.args.size(); j++) {
+        if (param_names.args[j] == kw_name) {
```

Comment:
Is the idea to specify "no keyword name for this parameter" as an empty string?  The user can still pass an empty-name parameter via kwargs `sorted(**{'':[]})` -- maybe we should add a `&& kw_name.size()` check somewhere in here?